File: permission.rb
# frozen_string_literal: true

require_relative 'lib/github_org_permission_checker'
require_relative 'lib/github_org_permission_notifier'

notifier = GithubOrgPermissionNotifier.new(
  webhook_url: ENV['SLACK_WEBHOOK_URL'],
  channel:     ENV['SLACK_CHANNEL_FOR_PERMISSION'],
  icon_emoji:  ENV['SLACK_ICON_EMOJI'],
  title:       ENV['SLACK_TITLE_FOR_PERMISSION']
)

GithubOrgPermissionChecker.new(
  org_name:         ENV['GITHUB_ORGANIZATION'],
  access_token:     ENV['GITHUB_ACCESS_TOKEN'],
  teams_permission: ENV['TEAMS_PERMISSION'],
  skip_days:        ENV['SKIP_DAYS'],
  notifier:         notifier
).execute

File: permission.go
package main

import (
	"os"
)

/** permission サブコマンド用の実装 **/
type Permission struct{}

func (f *Permission) Help() string {
	return "app permission"
}

func (f *Permission) Run(args []string) int {
	p := NewPermissionChecker(
		PermissionCheckerParams{
			orgName:         os.Getenv("GITHUB_ORGANIZATION"),
			accessToken:     os.Getenv("GITHUB_ACCESS_TOKEN"),
			teamsPermission: os.Getenv("TEAMS_PERMISSION"),
			skipDays:        os.Getenv("SKIP_DAYS"),
		},
	)
	p.Execute()
	return 0
}

func (f *Permission) Synopsis() string {
	return "Print \"Permission!\""
}
